app/models/concerns/host_params.rb

```diff
@@ -0,0 +1,83 @@
+module HostParams
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :host_parameters, :dependent => :destroy, :foreign_key => :reference_id, :inverse_of => :host
+    has_many :parameters, :dependent => :destroy, :foreign_key => :reference_id, :class_name => "HostParameter"
+    accepts_nested_attributes_for :host_parameters, :allow_destroy => true
+    include ParameterValidators
+    attr_reader :cached_host_params
+
+    def params
+      host_params.update(lookup_keys_params)
+    end
+
+    def clear_host_parameters_cache!
+      @cached_host_params = nil
+    end
+
+    def host_inherited_params(include_source = false)
+      hp = {}
+      params = host_inherited_params_objects
+      params.each do |param|
+        source = param_source(param.class.to_s.sub('Parameter', ''))
+        options = {:value => param.value,
+                   :source => source.to_sym,
+                   :safe_value => param.safe_value }
+        if source !=  'common'
+          options.merge!(:source_name => param.send(source).to_label)
+        end
+        hp.update(Hash[param.name => include_source ? options : param.value])
+      end
+      hp
+    end
+
+    def param_source(param_type)
+      case param_type
+        when 'Common'
+          N_('common')
+        when 'Os'
+          N_('operatingsystem')
```

Comment:
`source`  is also being used in: `param.send(source)` so I'll change it but then I need to handle those two separately so I'm putting back the `case` here.